src/Draco.Compiler.Tests/Semantics/TypeCheckingTests.cs

```diff
@@ -2248,4 +2248,43 @@ public void StringAssignableToObject()
         // Assert
         Assert.Empty(diags);
     }
+
+    [Fact]
+    public void InferredArrayElementTypeFromUsage()
+    {
+        // func main() {
+        //     val a = Array(5);
+        //     a[0] = 1;
+        // }
+
+        var main = SyntaxTree.Create(CompilationUnit(FunctionDeclaration(
```

Comment:
Why don't we parse the text directly instead of creating manually a SyntaxTree ?

Reply:
Test orthogonality, discussed